python/composio/tools/local/mathematical/actions/calculator.py

```diff
@@ -19,12 +21,54 @@ class CalculatorResponse(BaseModel):
 
 class Calculator(LocalAction[CalculatorRequest, CalculatorResponse]):
     """
-    Useful to perform any mathematical calculations, like sum, minus, multiplication, division, etc.
+    Performs mathematical calculations such as addition, subtraction, multiplication, and division.
     """
 
     _tags = ["calculator"]
 
+    # Define supported operators
+    operators = {
+        ast.Add: operator.add,
+        ast.Sub: operator.sub,
+        ast.Mult: operator.mul,
+        ast.Div: operator.truediv,
+        ast.Pow: operator.pow,
+        ast.USub: operator.neg,
+        ast.UAdd: operator.pos,
+    }
+
     def execute(self, request: CalculatorRequest, metadata: Dict) -> CalculatorResponse:
-        return CalculatorResponse(
-            result=str(eval(request.operation))  # pylint: disable=eval-used
-        )
+        try:
+            # Parse the expression into an AST
+            node = ast.parse(request.operation, mode="eval").body
+            result = self._safe_eval(node)
+            return CalculatorResponse(result=str(result))
+        except Exception as e:
```

Comment:
Good addition of error handling. This will make debugging easier and provide more informative responses to users when invalid expressions are entered. Consider adding more specific error messages for different types of exceptions to further improve user experience.